tests/test_execution_client.cpp

```diff
@@ -0,0 +1,28 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "sm_tests.h"
+
+namespace ara::exec {
+    /* Mock ExecutionClient for unit testing. */
+    class MockExecutionClient : public ExecutionClient {
+    public:
+        MOCK_METHOD(void, ReportApplicationState, (sm::FunctionGroupStateType state));
+    };
+}
+
+TEST_F(smTests,  testExecutionClient)
+{
+    ara::exec::MockExecutionClient mockEC;
+    mySM.executionClient = &mockEC;
+    EXPECT_CALL(mockEC,
+                ReportApplicationState(ara::sm::FunctionGroupStateType::On)).Times(testing::AnyNumber());
+
+    mySM.stateClient->SmSetState(ara::sm::FunctionGroupStateType::On);
+
+    // let changes to be applied
+    using namespace std::chrono_literals;
+    std::this_thread::sleep_for(500ms);
+
+    // due to possible GTest stuck while expecting a call
+    mySM.Kill();
```

Comment:
are u sure that's needed? we are killing SM in teardown function

Reply:
I know, but it is needed here. Without it, EC tests don't work XD